code/include/playfab/PlayFabProfilesDataModels_c.h

```diff
@@ -0,0 +1,259 @@
+#pragma once
+
+#include <playfab/PlayFabBaseModel_c.h>
+
+#pragma push_macro("IN")
+#undef IN
+
+// Profiles enums
+enum class PlayFabProfilesEffectType : uint32_t
+{
+    Allow,
+    Deny
+};
+
+enum class PlayFabProfilesOperationTypes : uint32_t
+{
+    Created,
+    Updated,
+    Deleted,
+    None
+};
+
+
+// Profiles structs
+typedef struct PlayFabProfilesEntityDataObject
+{
+    PlayFabJsonObject dataObject;
+    const char* escapedDataObject;
+    const char* objectName;
+
+} PlayFabProfilesEntityDataObject;
+
+typedef struct PlayFabProfilesEntityKey
+{
+    const char* id;
+    const char* type;
+
+} PlayFabProfilesEntityKey;
+
+typedef struct PlayFabProfilesEntityLineage
+{
+    const char* characterId;
+    const char* groupId;
+    const char* masterPlayerAccountId;
+    const char* namespaceId;
+    const char* titleId;
+    const char* titlePlayerAccountId;
+
+} PlayFabProfilesEntityLineage;
+
+typedef struct PlayFabProfilesEntityPermissionStatement
+{
+    const char* action;
+    const char* comment;
+    PlayFabJsonObject condition;
+    PlayFabProfilesEffectType effect;
+    PlayFabJsonObject principal;
+    const char* resource;
+
+} PlayFabProfilesEntityPermissionStatement;
+
+typedef struct PlayFabProfilesEntityProfileFileMetadata
+{
+    const char* checksum;
+    const char* fileName;
+    time_t lastModified;
+    int32_t size;
+
+} PlayFabProfilesEntityProfileFileMetadata;
+
+typedef struct PlayFabProfilesEntityStatisticChildValue
+{
+    const char* childName;
+    const char* metadata;
+    int32_t value;
+
+} PlayFabProfilesEntityStatisticChildValue;
+
```

Comment:
nit: can you remove the blank line before the end of each struct?  it'll make it look better

Reply:
ya, thats an autogen issue.  Will fix that

---
In reply to: [604532683](https://github.com/jplafonta/XPlatCSdk/pull/1#discussion_r604532683) [](ancestors = 604532683)